File: Homework/4.3.1.py
# Excercise 4.3.1.py #

def IsPalindrome(string):
    l = len(string)
    if l <= 0:
        return False
    if l % 2:
        string = string.replace(string[(l-1)//2], '')
    stack = []
    stack.append(string[0])
    for c in string[1:]:
        if c != stack[-1]:
            stack.append(c)
        else:
            stack.pop()
    print(stack)
    return len(stack) == 0

File: FDSC/Homework/4.3.1.c
/* Excercise 4.3.1.c */
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdbool.h>

bool isPalindrome(char *s);

int main(void)
{
    char s[10] = "abcdedcba";
    if (isPalindrome(s))
        printf("IS PALINDROME!\n");
    else
        printf("No way!\n");
    
    return 0;
}

bool isPalindrome(char *s)
{
	if(s==NULL || !strcmp(s,""))
		return true;

	for(char* p = s + strlen(s) -1; p>s;)
		if(isalnum(*s) && isalnum(*p) &&
            tolower(*(s++)) != tolower(*(p--)))
			return false;
    	else if(!isalnum(*s))
    		s++;
    	else if(!isalnum(*p))
    		p--;
    return true;
}
